### PR TITLE
feat: implement redirect logic

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: ldi-core/request-executor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/requestexecutor/executor/noauth/DefaultConfig.java [hunk 1]
```diff
@@ -10,14 +10,20 @@
...
+	public DefaultConfig(Collection<Header> headers, boolean enableRedirectHandling) {
 		this.headers = headers;
-	}
+        this.enableRedirectHandling = enableRedirectHandling;
+    }
 
 	public RequestExecutor createRequestExecutor() {
-		return new DefaultRequestExecutor(
-				HttpClientBuilder.create().setDefaultHeaders(headers).disableRedirectHandling().build());
+		HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
```
Comment (jobulcke) [inline ldi-core/request-executor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/requestexecutor/executor/noauth/DefaultConfig.java:21]: Could this be final please?

File: ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClientProcessor.java [hunk 2]
```diff
@@ -126,7 +127,7 @@ private Retry getRetry(final ProcessContext context) {
 
 	private RequestExecutor getRequestExecutor(final ProcessContext context) {
 		return switch (getAuthorizationStrategy(context)) {
-			case NO_AUTH -> requestExecutorFactory.createNoAuthExecutor();
+			case NO_AUTH -> requestExecutorFactory.createNoAuthExecutor(new ArrayList<>());
```
Comment (jobulcke) [inline ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClientProcessor.java:130]: Why is this empty list suddenly added as a param? Doesn't the original line do the same anymore?

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client-connector/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioLdesClientConnectorAutoConfig.java [hunk 3]
```diff
@@ -40,8 +42,8 @@ public static class LdioClientConnectorConfigurator implements LdioInputConfigur
 		public static final String PROXY_URL_REPLACEMENT = "proxy-url-replacement";
 		private final ApplicationEventPublisher eventPublisher;
 		private final ObservationRegistry observationRegistry;
-		private final RequestExecutorFactory requestExecutorFactory = new RequestExecutorFactory();
-		private final RequestExecutor baseRequestExecutor = requestExecutorFactory.createNoAuthExecutor();
+		private final RequestExecutorFactory requestExecutorFactory = new RequestExecutorFactory(false);
+		private final RequestExecutor baseRequestExecutor = requestExecutorFactory.createNoAuthExecutor(new ArrayList<>());
```
Comment (jobulcke) [inline ldi-orchestrator/ldio-connectors/ldio-ldes-client-connector/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioLdesClientConnectorAutoConfig.java:46]: Why is this empty list suddenly added as a param? Doesn't the original line do the same anymore?

**Review by jobulcke — APPROVED** [review]

File: ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClientProcessor.java (hunk 2 above)
Comment (Tomvbe) [reply ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClientProcessor.java]: Same story as this morning, I did something else first when this made sense, reverted the changes and forgot to revert this :smile_cat:

File: ldi-orchestrator/ldio-connectors/ldio-ldes-client-connector/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioLdesClientConnectorAutoConfig.java (hunk 3 above)
Comment (Tomvbe) [reply ldi-orchestrator/ldio-connectors/ldio-ldes-client-connector/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioLdesClientConnectorAutoConfig.java]: Same story as this morning, I did something else first when this made sense, reverted the changes and forgot to revert this :smile_cat:

**Pull request merged**